scene/animation/tween.cpp

```diff
@@ -581,7 +586,23 @@ bool PropertyTweener::step(double &r_delta) {
 
 	double time = MIN(elapsed_time - delay, duration);
 	if (time < duration) {
-		target_instance->set_indexed(property, tween->interpolate_variant(initial_val, delta_val, time, duration, trans_type, ease_type));
+		if (custom_method.is_valid()) {
+			const Variant t = tween->interpolate_variant(0.0, 1.0, time, duration, trans_type, ease_type);
+			const Variant *argptr = &t;
+
+			Variant result;
+			Callable::CallError ce;
+			custom_method.callp(&argptr, 1, result, ce);
+			if (ce.error != Callable::CallError::CALL_OK) {
+				ERR_FAIL_V_MSG(false, "Error calling custom method from PropertyTweener: " + Variant::get_callable_error_text(custom_method, &argptr, 1, ce) + ".");
+			} else if (result.get_type() != Variant::FLOAT) {
+				ERR_FAIL_V_MSG(false, vformat("Wrong return type in PropertyTweener custom method. Expected float, got %s.", Variant::get_type_name(result.get_type())));
+			}
+
+			target_instance->set_indexed(property, Animation::interpolate_variant(initial_val, final_val, result));
+		} else {
+			target_instance->set_indexed(property, tween->interpolate_variant(initial_val, delta_val, time, duration, trans_type, ease_type));
+		}
```

Comment:
Can you make it to pass custom_interpolator as an argument to `run_equation()` and have it available as a static function? Like:

```cpp
real_t Tween::run_equation(TransitionType p_trans_type, EaseType p_ease_type, Callable &p_method, real_t p_time, real_t p_initial, real_t p_delta, real_t p_duration) {
	if (p_duration == 0) {
		// Special case to avoid dividing by 0 in equations.
		return p_initial + p_delta;
	}

	interpolater func = interpolaters[p_trans_type][p_ease_type];
	real_t result = func(p_time, p_initial, p_delta, p_duration);
	Callable::CallError ce;
	custom_method.callp(&argptr, 1, result, ce);

...
...

	return result;
}
```

Since tween interpolation is reused in the AnimationTrackEditor's Bake function and the planned capture function in https://github.com/godotengine/godot/pull/86715, it would be ideal if custom_interpolator could be used there.

But well, since the AnimationTrackEditor needs a UI to implement curves, so we may not have to worry about the Bake functionality for now.

Reply:
We agreed that we can add a method that uses Struct later if necessary (maybe after https://github.com/godotengine/godot/pull/82155 ?).